test/parallel/test-async-wrap-asyncresource-constructor.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+require('../common');
+
+// This tests that AsyncResource throws an error if bad parameters are passed
+
+const assert = require('assert');
+const AsyncResource = require('async_hooks').AsyncResource;
+const async_wrap = process.binding('async_wrap');
+
+/* without this, there is nothing to do,
+* and the constructor would return immediately
+*/
+async_wrap.async_hook_fields[async_wrap.constants.kInit] = 1;
```

Comment:
I'd change it and wait for the fix. If there's already a PR w/ the fix then we'll get it in quick. If not then I'll make one. Either way it's a simple fix that shouldn't take more than a day to get landed.